src/System.Management.Automation/security/Win32Native/WinTrust.cs

```diff
@@ -0,0 +1,434 @@
+        public CRYPT_ATTR_BLOB sEncodedMemberInfo;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct CRYPTCATATTRIBUTE
+    {
+        public uint cbStruct;
+        [MarshalAs(UnmanagedType.LPWStr)] public string pwszReferenceTag;
+        public uint dwAttrTypeAndAction;
+        public uint cbValue;
+        public IntPtr pbValue;
+        public uint dwReserved;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct CRYPTCATSTORE
+    {
+        public uint cbStruct;
+        public uint dwPublicVersion;
+        [MarshalAs(UnmanagedType.LPWStr)] public string pwszP7File;
+        public IntPtr hProv;
+        public uint dwEncodingType;
+        public uint fdwStoreFlags;
+        public IntPtr hReserved;
+        public IntPtr hAttrs;
+        public IntPtr hCryptMsg;
+        public IntPtr hSorted;
+    }
+
+    [StructLayoutAttribute(LayoutKind.Sequential)]
+    internal struct WINTRUST_DATA
+    {
+        public uint cbStruct;
+        public IntPtr pPolicyCallbackData;
+        public IntPtr pSIPClientData;
+        public WinTrustUIChoice dwUIChoice;
+        public uint fdwRevocationChecks;
+        public WinTrustUnionChoice dwUnionChoice;
+        public unsafe void* pChoice;
+        public WinTrustAction dwStateAction;
+        public IntPtr hWVTStateData;
+        public IntPtr pwszURLReference;
+        public WinTrustProviderFlags dwProvFlags;
+        public uint dwUIContext;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct WINTRUST_FILE_INFO
+    {
+        public uint cbStruct;
+        public unsafe char* pcwszFilePath;
+        public IntPtr hFile;
+        public IntPtr pgKnownSubject;
+    }
+
+    [StructLayoutAttribute(LayoutKind.Sequential)]
+    internal struct WINTRUST_BLOB_INFO
+    {
+        public uint cbStruct;
+        public Guid gSubject;
+        public unsafe char* pcwszDisplayName;
+        public uint cbMemObject;
+        public unsafe byte* pbMemObject;
+        public uint cbMemSignedMsg;
+        public IntPtr pbMemSignedMsg;
+    }
+
+    [DllImport(WinTrustDll, CharSet = CharSet.Unicode, EntryPoint = "CryptCATAdminAcquireContext2",
+        SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool NativeCryptCATAdminAcquireContext2(
+        out SafeCATAdminHandle phCatAdmin,
+        IntPtr pgSubsystem,
+        [MarshalAs(UnmanagedType.LPWStr)] string pwszHashAlgorithm,
+        IntPtr pStrongHashPolicy,
+        uint dwFlags
+    );
+
+    internal static SafeCATAdminHandle CryptCATAdminAcquireContext2(string hashAlgorithm)
+    {
+        if (!NativeCryptCATAdminAcquireContext2(out var adminHandle, IntPtr.Zero, hashAlgorithm, IntPtr.Zero, 0))
+        {
+            throw new Win32Exception();
+        }
+
+        return adminHandle;
+    }
+
+    [DllImport(WinTrustDll, CharSet = CharSet.Unicode, EntryPoint = "CryptCATAdminCalcHashFromFileHandle2",
+        SetLastError = true)]
```

Comment:
Cases like this where properties or parameters are left on multiple lines will cause `CodeFactor` to emit errors. Can you please update cases like this one to put them either on the same line, or having each item on its own line?

Reply:
Sure, I'll go for multiple lines as entries like this got a bit too long.